### PR TITLE
feat(dms): Add examples

### PR DESCRIPTION
**Pull request by dmicheneau**
### Description of your changes

Add DMS kafka instance example
Add secret password for manager and user

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.
- [x] test and vaildate dms kafka instance
- [x] test and vaildate dms kafka topic
- [x] test and vaildate dms kafka user

### How has this code been tested
```
NAME                                                              READY   SYNCED   EXTERNAL-NAME                                                  AGE
kafkauser.dms.flexibleengine.upbound.io/example-dms-kafka-user1   True    True     0234322a-1349-4ce8-811c-ec08cf2f6a93/example-dms-kafka-user1   82s

NAME                                                               READY   SYNCED   EXTERNAL-NAME             AGE
kafkatopic.dms.flexibleengine.upbound.io/example-dms-kafka-topic   True    True     example-dms-kafka-topic   16m

NAME                                                                     READY   SYNCED   EXTERNAL-NAME                          AGE
kafkainstance.dms.flexibleengine.upbound.io/example-dms-kafka-instance   True    True     0234322a-1349-4ce8-811c-ec08cf2f6a93   108m

NAME                                                    READY   SYNCED   EXTERNAL-NAME     AGE
keypair.ecs.flexibleengine.upbound.io/example-keypair   True    True     example-keypair   4d22h

NAME                                            READY   SYNCED   EXTERNAL-NAME                          AGE
vpc.vpc.flexibleengine.upbound.io/example-vpc   True    True     13532b25-8836-46d7-bc57-14ace2bd47df   6d3h

NAME                                                           READY   SYNCED   EXTERNAL-NAME                          AGE
securitygroup.vpc.flexibleengine.upbound.io/example-secgroup   True    True     e71c8f48-66d9-4113-b5ed-b653ecf6b731   4h20m

NAME                                                     READY   SYNCED   EXTERNAL-NAME                          AGE
vpcsubnet.vpc.flexibleengine.upbound.io/example-subnet   True    True     2e044c75-c21f-4bad-9c04-c85a431f536e   6d3h

```

### CODE REVIEW

File: config/overrides.go [hunk 1]
```diff
@@ -58,8 +58,14 @@ func KnownReferencers() config.ResourceOption { //nolint:gocyclo
...
-				r.References[k] = config.Reference{
-					Type: tools.GenerateType("ecs", "Instance"),
+				if r.ShortGroup == "dms" {
+					r.References[k] = config.Reference{
+						Type: tools.GenerateType("dms", "KafkaInstance"),
+					}
+				} else {
+					r.References[k] = config.Reference{
+						Type: tools.GenerateType("ecs", "Instance"),
+					}
```
Comment (azrod) [inline config/overrides.go:68]: it's better to use a switch to handle future cases

```
switch r.ShortGroup {
case "dms":
	r.References[k] = config.Reference{
		Type: tools.GenerateType("dms", "KafkaInstance"),
	}
default:
	r.References[k] = config.Reference{
		Type: tools.GenerateType("ecs", "Instance"),
	}
}
```

File: examples/dms/kafkauser_secret.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    testing.upbound.io/example-name: example_dms_kafka_secret_user
+  name: example-dms-kafka-secret-user
+  namespace: crossplane-system
+type: Opaque
+stringData:
+  example-dms-kafka-key-user1: SuperS3cr3tP4ssw0rdUSER1
```
Comment (azrod) [inline examples/dms/kafkauser_secret.yaml:10]: Use the password key for more understanding

File: examples/dms/kafkainstance_secret.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+kind: Secret
+metadata:
+  labels:
+    testing.upbound.io/example-name: example_dms_kafka_secret
+  name: example-dms-kafka-secret
+  namespace: crossplane-system
+type: Opaque
+stringData:
+  example-dms-kafka-key-manager: SuperS3cr3tP4ssw0rdMANAGER
+  example-dms-kafka-key-user: SuperS3cr3tP4ssw0rdUSER
```
Comment (azrod) [inline examples/dms/kafkainstance_secret.yaml:11]: Use `password-manager`and `password-user` for more understanding

File: examples/dms/kafkauser.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+apiVersion: dms.flexibleengine.upbound.io/v1beta1
+kind: KafkaUser
+metadata:
+  annotations:
+    meta.upbound.io/example-id: dms/v1beta1/kafkauser
+  labels:
+    testing.upbound.io/example-name: example_dms_kafka_user1
+  name: example-dms-kafka-user1
```
Comment (azrod) [inline examples/dms/kafkauser.yaml:8]: Why example_dms_kafka_user`1` and example-dms-kafka-user`1` ?

File: examples/dms/kafkauser.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  forProvider:
+    instanceIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: example_dms_kafka_instance
+    name: example-dms-kafka-user1
```
Comment (azrod) [inline examples/dms/kafkauser.yaml:14]: And here

File: examples/dms/kafkauser.yaml (hunk 4 above)
Comment (dmicheneau) [reply examples/dms/kafkauser.yaml:8]: to avoid any confusion with the user declare in kafka instance

**Review by azrod — COMMENTED** [review]
Please clean your commit msg

**Review by azrod — APPROVED** [review]

**Pull request merged**